src/Microsoft.DotNet.Cli.Utils/CommandResolution/DefaultCommandResolver.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.Extensions.PlatformAbstractions;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    public class DefaultCommandResolver : ICommandResolver
+    {
+        private IList<ICommandResolver> _orderedCommandResolvers;
+
+        public DefaultCommandResolver() : this(new PackagedCommandSpecCreator()) { }
+
+        public DefaultCommandResolver(IPackagedCommandSpecCreator packagedCommandSpecCreator)
```

Comment:
Maybe move this to an extension method. The idea is, if this changes, we don't need to change this class. I am not sure where this is used, like the resolvers and all, otherwise, I would suggest that the "Program.cs" that use this should be the ones setting it up.